### PR TITLE
Introduce single page apps.

### PR DESCRIPTION
**Pull request by dfitchett**
## Summary

Set up structure for single page apps.

Associated tickets or Slack threads:
- https://github.com/department-of-veterans-affairs/abd-vro/issues/3554

## How to test this PR
1. Pull branch
2. from root dir, run `./gradlew vro-streamlit:check` to run all checks/lints/unit tests.
3. from root dir, run `./gradlew :vro-streamlit:docker ` then run `COMPOSE_PROFILES="streamlit" ./gradlew :dockerComposeUp` to run the container. Then open browser to http://0.0.0.0:8501/ to view webpages.

### CODE REVIEW

**Review by brostk — APPROVED** [review]
LGTM

File: vro-streamlit/src/vro_streamlit/main.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                st.markdown('Authorized', help='User authorization status')
+            with col2:
+                st.markdown(f'`{config.ENV}`')
+                st.markdown(':large_green_circle:' if st.session_state.database_connected else ':red_circle:', unsafe_allow_html=True)
+                st.markdown(':large_green_circle:' if st.session_state.user else ':red_circle:', unsafe_allow_html=True)
+
+        if st.session_state.user:
+            st.button('Log Out', use_container_width=True, on_click=update_login_status, key=LOGOUT_BUTTON)
+        else:
+            st.button('Log In', use_container_width=True, on_click=update_login_status, key=LOGIN_BUTTON)
```
Comment (gabezurita) [inline vro-streamlit/src/vro_streamlit/main.py]: The above could be made more DRY and concise by doing the following (this is not tested):
```python
        button_text = 'Log Out' if st.session_state.user else 'Log In'
        st.button(button_text, on_click=update_login_status, use_container_width=True)
```

File: vro-streamlit/src/vro_streamlit/auth/auth_service.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from vro_streamlit.auth.user import User
+
+
+def log_in() -> User:
+    return User('test')
+
+
+def log_out() -> bool:
+    return True
```
Comment (gabezurita) [inline vro-streamlit/src/vro_streamlit/auth/auth_service.py:9]: I would maybe add TODOs here:
```
# auth_service.py
def log_in() -> User:
    # TODO: Implement real authentication logic
    pass

def log_out() -> bool:
    # TODO: Implement real logout logic
    pass
```

File: vro-streamlit/test/directory/test_home.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    app_test.session_state.user = User(USERNAME)
+    app_test.run()
+    assert not app_test.exception
+    assert_markdown_contains_values(app_test.markdown, f'Hello, {USERNAME}!')
+    assert_button_contains_label(app_test.button[0], 'Log Out')
+
+    # Click page which reloads page
+    app_test.button(key=LOGIN_BUTTON).click().run()
+    assert_markdown_contains_values(app_test.markdown, 'Please Login')
+    assert_button_contains_label(app_test.button[0], 'Log In')
```
Comment (gabezurita) [inline vro-streamlit/test/directory/test_home.py:49]: Eventually, we'll want to test for session persistence across reloads:
```python
# test_directory/test_home.py

def test_session_persistence(app_test) -> None:
    # Simulate user login
    app_test.run()
    assert_no_exceptions(app_test)
    app_test.button(key=LOGIN_BUTTON).click().run()
    assert_no_exceptions(app_test)
    assert_markdown_contains_values(app_test.markdown, f'Hello, {USERNAME}!')

    # Simulate page reload
    app_test.run()
    assert_no_exceptions(app_test)
    assert_markdown_contains_values(app_test.markdown, f'Hello, {USERNAME}!')  # Session should persist

```

File: vro-streamlit/src/vro_streamlit/directory/home.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    st.subheader('Welcome to the home page!')
+
+    user_logged_in = 'user' in st.session_state and st.session_state.user is not None
+
+    msg = 'Please Login' if not user_logged_in else f'Hello, {st.session_state.user.username}!'
+    st.write(msg)
+
+    button_text = 'Log Out' if user_logged_in else 'Log In'
+
+    st.button(button_text, key=LOGIN_BUTTON, on_click=partial(update_login_status, user_logged_in))
```
Comment (gabezurita) [inline vro-streamlit/src/vro_streamlit/directory/home.py]: Is the `and st.session_state.user is not None` in line 26 ` user_logged_in = 'user' in st.session_state and st.session_state.user is not None` redundant?

An alternative structure (maybe more readable)?
```python
def show() -> None:
    col1, col2 = st.columns([0.04, 0.96])
    col1.image(LOGO, width=100)
    col2.header('Home')
    st.subheader('Welcome to the home page!')

    user = st.session_state.get('user')
    if user:
        st.write(f'Hello, {user.username}!')
        st.button('Log Out', key=LOGIN_BUTTON, on_click=update_login_status)
    else:
        st.write('Please Login')
        st.button('Log In', key=LOGIN_BUTTON, on_click=update_login_status)

```

**Review by gabezurita — APPROVED** [review]
LGTM! Some alternatives

File: vro-streamlit/src/vro_streamlit/auth/auth_service.py (hunk 2 above)
Comment (dfitchett) [reply vro-streamlit/src/vro_streamlit/auth/auth_service.py:9]: I would! But I'm almost ready to push another PR that does just this! 😬

File: vro-streamlit/test/directory/test_home.py (hunk 3 above)
Comment (dfitchett) [reply vro-streamlit/test/directory/test_home.py:49]: Agreed! There's some gotcha involved with doing that, as streamlit sessions are shortlived, and a page refresh currently resets that session.

**Pull request merged**